### ISSUE TITLE
PQN's Q(lambda) target caches Q at the wrong state (lands-in, not starts-from), diverging from the algorithm it's adapted from

### ISSUE THREAD

**Issue by epignatelli** [keraJLi/rejax#34]
## What's wrong

`PQN.calculate_targets` in `rejax/algos/pqn.py` caches the wrong Q-value per transition when computing the Q(λ) bootstrap target, diverging from both the correct recursion and from the official implementation this was adapted from ([`mttga/purejaxql`](https://github.com/mttga/purejaxql), credited in this file's own docstring).

`Trajectory.next_q` is populated in `collect_trajectories` as:

```python
next_q = self.agent.apply(ts.q_ts.params, next_obs)
...
transition = Trajectory(ts.last_obs, action, next_q, reward, done)
```

i.e. `next_q[i] = Q(s_{i+1})` - the Q-value at the state the transition *lands in*. `mttga/purejaxql`'s own `_step_env` instead stores `q_val=q_vals`, computed from `last_obs` *before* stepping - i.e. `q_val[i] = Q(s_i)`, the state the transition *starts from*. That's not a cosmetic rename: `calculate_targets`'s backward scan threads whichever of these gets returned from each step forward into the *next* (earlier-index) step's one-step bootstrap, and only `Q(s_i)` is the value that step actually needs.

(Separately, I believe `purejaxql`'s own version - and by inheritance this one - has an *additional*, narrower bug in how the scan's carry is seeded, filed against that repo: mttga/purejaxql#12. This issue is about the extra divergence unique to this repo's adaptation, on top of that inherited one.)

## Minimal reproduction

Self-contained (`pip install jax rlax`), no repo checkout needed - `calculate_targets` below is copied verbatim from `rejax/algos/pqn.py`, only de-classed into a free function:

```python
from typing import NamedTuple
import jax, jax.numpy as jnp, numpy as np, rlax

class Trajectory(NamedTuple):
    reward: jnp.ndarray
    done: jnp.ndarray
    next_q: jnp.ndarray

def calculate_targets(trajectories, max_last_q, gamma, td_lambda):
    def get_target(lambda_return_and_next_q, t):
        lambda_return, next_q = lambda_return_and_next_q
        return_bootstrap = next_q + td_lambda * (lambda_return - next_q)
        lambda_return = t.reward + (1 - t.done) * gamma * (return_bootstrap)
        max_next_q = t.next_q.max(axis=1)
        return (lambda_return, max_next_q), lambda_return

    max_last_q = jnp.where(trajectories.done[-1], 0, max_last_q)
    lambda_returns = trajectories.reward[-1] + gamma * max_last_q
    _, targets = jax.lax.scan(
        get_target, (lambda_returns, max_last_q),
        Trajectory(trajectories.reward[:-1], trajectories.done[:-1], trajectories.next_q[:-1]),
        reverse=True,
    )
    return jnp.concatenate((targets, lambda_returns[None]))

# a fixed "true" Q(s_i) for a 6-step, single-env, no-termination trajectory
rng = np.random.default_rng(1)
T = 6
Q = rng.normal(size=(T + 1,)).astype(np.float32) * 3       # Q(s_0) .. Q(s_T)
reward = rng.normal(size=(T,)).astype(np.float32)
done = np.zeros(T, dtype=np.float32)
gamma, lam = 0.9, 0.8

next_q = jnp.asarray(Q[1:])[:, None, None]                  # this repo's own field convention: next_q[i] = Q(s_{i+1})
trajectories = Trajectory(jnp.asarray(reward)[:, None], jnp.asarray(done)[:, None], next_q)
targets = np.asarray(calculate_targets(trajectories, jnp.asarray([Q[T]]), gamma, lam)).ravel()

# ground truth: rlax.lambda_returns, DeepMind's independently-implemented
# Q(lambda) (its docstring documents this exact setup: "Q(lambda): v_t = max(q_t, axis=-1)")
next_values = jnp.asarray(Q[1:])
discount = gamma * (1.0 - jnp.asarray(done))
ground_truth = np.asarray(rlax.lambda_returns(jnp.asarray(reward), discount, next_values, lam))

print("rejax targets:     ", targets)
print("rlax ground truth: ", ground_truth)
print("abs diff:           ", np.abs(targets - ground_truth))
```

Output:

```
rejax targets:      [ 0.6678596  -0.1273534   0.29413748 -0.6790099  -1.3173257  -2.1862278 ]
rlax ground truth:  [ 1.1852053   0.22279646 -0.4447388  -0.04884233 -0.78632873 -2.1862278 ]
abs diff:            [0.5173457  0.35014987 0.7388763  0.6301676  0.530997   0.        ]
```

For comparison, `purejaxql`'s own version on the identical ground truth (same `Q`/`reward`/`done`, only re-packaged into *its* `q_val[i] = Q(s_i)` convention) gets closer but is still off:

```
purejaxql targets:  [ 1.0425059   0.02460287 -0.72000766 -0.43116036 -1.3173257  -2.1862278 ]
abs diff:            [0.14269948 0.19819358 0.27526885 0.38231802 0.530997   0.        ]
```

## Suggested fix

Cache the Q-value at the state each transition *starts from* (`Q(s_i)`, matching `mttga/purejaxql`'s `q_val`), not the state it lands in (`Q(s_{i+1})`), before feeding it through the same backward-scan recursion.

Happy to be told I'm missing something about the intended semantics - but the divergence against `rlax.lambda_returns` (whose docstring documents this exact target as "Peng's Q(lambda)") seems hard to explain otherwise, and it's a bigger effect than the inherited seeding issue alone.
